packages/devtools/lib/src/utils.dart

```diff
@@ -117,6 +117,28 @@ void executeWithDelay(Duration delay, void callback(),
   }
 }
 
+String longestFittingSubstring(
+  String originalText,
+  num maxWidth,
+  List<num> asciiMeasurements,
+) {
+  if (originalText.isEmpty) return originalText;
+
+  num currentWidth = 0;
+  int longestIndex = 0;
+  final charCodes = ascii.encode(originalText);
+  for (int i = 0; i < charCodes.length; i++) {
```

Comment:
comment that this is only approximate due to ignoring kerning.

Reply:
done.